cmd/tomlfileconnection/tomlfileconnection.go

```diff
@@ -0,0 +1,58 @@
+// Example: How to connect to the server with the toml file configuration
+// Prerequiste: following the Snowflake doc: https://docs.snowflake.com/en/developer-guide/snowflake-cli-v2/connecting/specify-credentials
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	sf "github.com/snowflakedb/gosnowflake"
+)
+
+func main() {
+	if !flag.Parsed() {
+		flag.Parse()
+	}
+
+	os.Setenv("SNOWFLAKE_HOME", "<The directory path where the toml file exists>")
+	os.Setenv("SNOWFLAKE_DEFAULT_CONNECTION_NAME", "<DSN Name>")
+
+	cfg, err := sf.LoadConnectionConfig()
```

Comment:
The main idea is to read the configuration automatically not by calling this method  I think that we can suggest driver to automatically read the toml configuration by specific value of parameter in method Open(dsn string). For example: when the dsn value will be set as "autoconfiguration" then the driver will automatically read the TOML file configuration. 
`func (d SnowflakeDriver) Open(dsn string) (driver.Conn, error) {
	ctx := context.Background()
if (dsn == 'autoconfiguration') {
    cfg, err := sf.LoadConnectionConfig()
} else {
	cfg, err := ParseDSN(dsn)
}
	if err != nil {
		return nil, err
	}
	return d.OpenWithConfig(ctx, *cfg)
}`

Reply:
I applied it. Instead of 'autoconfiguration', I used 'autoConfig'. Do you think 'autoconfiguration' is much better? If you share your idea, I will add a comment to cmd/tomlfileconfiguration. Thank you
'

Reply:
Let's stay with you proposal 'autoConfig'